CMakeLists.txt

```diff
@@ -0,0 +1,1109 @@
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/sysroot/include/wasi
+
+  # basic headers
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__errno.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__functions_malloc.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__functions_memcpy.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__header_inttypes.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__macro_PAGESIZE.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__struct_stat.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__struct_timespec.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_blkcnt_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_blksize_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_clock_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_dev_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_gid_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_ino_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_mode_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_nlink_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_off_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_ssize_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_suseconds_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_time_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/basics/include/__typedef_uid_t.h ${CMAKE_BINARY_DIR}/sysroot/include/
```

Comment:
You need to enumerate the files, yes, I can create a list of the files and use a loop, but, it will make things a bit more complicated due to the dependency tracking.

Reply:
Is there a way we can factor out some of the redundancy, with a macro or so? That would make this code easier to manage.

Reply:
I'll try to do something with a macro to help with this.

Reply:
Collapsed it into an invocation per directory.